content/docs/rfcs/51/README.md

```diff
@@ -0,0 +1,243 @@
+---
+slug: 51
+title: 51/WAKU2-RELAY-SHARDING
+name: Waku v2 Relay Sharding
+status: raw
+category: Standards Track
+tags:
+editor: Daniel Kaiser <danielkaiser@status.im>
+contributors:
+---
+
+# Abstract
+
+This document describes ways of sharding the [Waku relay](/spec/11/) topic,
+allowing Waku networks to scale in the number of content topics.
+
+> *Note*: Scaling in the size of a single content topic is out of scope for this document.
+
+# Background and Motivation
+
+[Unstructured P2P networks](https://en.wikipedia.org/wiki/Peer-to-peer#Unstructured_networks)
+are more robust and resilient against DoS attacks compared to
+[structured P2P networks](https://en.wikipedia.org/wiki/Peer-to-peer#Structured_networks)).
+However, they do not scale to large traffic loads.
+A single [libp2p gossipsub mesh](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.0.md#gossipsub-the-gossiping-mesh-router),
+which carries messages associated with a single pubsub topic, can be seen as a separate unstructured P2P network
+(gossip and control messages go beyond these boundaries, but at its core, it is a separate P2P network).
+With this, the number of [Waku relay](/spec/11/) content topics that can be carried over a pubsub topic is limited.
+This prevents app protocols that aim to span many multicast groups (realized by content topics) from scaling.
+
+This document specifies three pubsub topic sharding methods (with varying degrees of automation),
+which allow application protocols to scale in the number of content topics.
+This document also covers discovery of topic shards.
+
+# Named Sharding
+
+*Named sharding* offers apps to freely choose pubsub topic names.
+App protocols SHOULD follow the naming structure detailed in [23/WAKU2-TOPICS](/spec/23/).
+With named sharding, managing discovery falls into the responsibility of apps.
+
+The default Waku pubsub topic `/waku/2/default-waku/proto` can be seen as a named shard available to all app protocols.
+
+> *Note*: Future versions of this document are planned to give more guidance with respect to discovery via
+[33/WAKU2-DISCV5](/spec/33/),
+[DNS discovery](https://eips.ethereum.org/EIPS/eip-1459),
+and inter-mesh discovery via gossipsub control messages (also using circuit relay).
+It might make sense to deprecate [23/WAKU2-TOPICS](/spec/23/) as a separate spec and merge it here.
+
+From an app protocol point of view, a subscription to a content topic `waku2/xxx` on a shard named /mesh/v1.1.1/xxx would look like:
+
+`subscribe("/waku2/xxx", "/mesh/v1.1.1/xxx")`
+
+# Static Sharding
+
+*Static sharding* offers an array of shard clusters.
```

Comment:
Perhaps possible to be clearer here on an array of 64K shard clusters, consisting of 64 shards each? And perhaps mention the total number of available shards with static sharding. Wasn't immediately clear to me in terms of "index", "cluster" and "array".

Reply:
Sounds great. Thanks!